app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+
+    public void receive(int size) throws IOException {
+        if (size > 0) {
+            implicitHeaderMode();
+            setupBytesArray(size & 0xFF);
+            SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        } else {
+            explicitHeaderMode();
+            setupBytesArray(MODE_LONG_RANGE_MODE | MODE_RX_SINGLE);
+            SPIWrite(REG_OP_MODE, bytes);
+        }
+    }
+
+    private void idle() throws IOException {
+        setupBytesArray(MODE_LONG_RANGE_MODE | MODE_STDBY);
+        SPIWrite(REG_OP_MODE, bytes);
+    }
+
+    private void sleep() throws IOException {
+        setupBytesArray(MODE_LONG_RANGE_MODE | MODE_SLEEP);
+        SPIWrite(REG_OP_MODE, bytes);
+    }
+
+    private void setTxPower(int level, int pin) throws IOException {
+        if (pin == PA_OUTPUT_RFO_PIN) {
+            if (level < 0) {
+                level = 0;
+            } else if (level > 14) {
+                level = 14;
+            }
+            setupBytesArray(0x70 | level);
+            SPIWrite(REG_PA_CONFIG, bytes);
+        } else {
+            if (level < 2) {
+                level = 2;
+            } else if (level > 17) {
+                level = 17;
+            }
+            if (level == 17) {
+                Log.d(name, "max power output");
+                setupBytesArray(0x87);
+                SPIWrite(REG_PA_DAC, bytes);
+            } else {
+                setupBytesArray(0x84);
+                SPIWrite(REG_PA_DAC, bytes);
+                setupBytesArray(PA_BOOST | 0x70 | (level - 2));
+                SPIWrite(REG_PA_CONFIG, bytes);
+            }
+        }
+        Log.d(name, "Power " + SPIRead(REG_PA_CONFIG, 1).get(0));
+    }
+
+    private void setFrequency(int frq) throws IOException {
+        this.frequency = frq;
+        int frf = (frq << 19) / 32000000;
+        Log.d(name, "frf = " + frf);
+        Log.d(name, "freq = " + ((frf >> 16) & 0xFF) + " " + ((frf >> 8) & 0xFF) + " " + (frf & 0xFF));
+        setupBytesArray((frf >> 16) & 0xFF);
+        SPIWrite(REG_FRF_MSB, bytes);
+        setupBytesArray((frf >> 8) & 0xFF);
+        SPIWrite(REG_FRF_MID, bytes);
+        setupBytesArray(frf & 0xFF);
+        SPIWrite(REG_FRF_LSB, bytes);
+    }
+
+    private void setSpreadingFactor(int spreadingFactor) throws IOException {
+        if (spreadingFactor < 6) {
+            spreadingFactor = 6;
+        } else if (spreadingFactor > 12) {
+            spreadingFactor = 12;
+        }
+
+        if (spreadingFactor == 6) {
+            setupBytesArray(0xc5);
+            SPIWrite(REG_DETECTION_OPTIMIZE, bytes);
+            setupBytesArray(0x0c);
+            SPIWrite(REG_DETECTION_THRESHOLD, bytes);
+        } else {
+            setupBytesArray(0xc3);
+            SPIWrite(REG_DETECTION_OPTIMIZE, bytes);
+            setupBytesArray(0x0a);
+            SPIWrite(REG_DETECTION_THRESHOLD, bytes);
+            setupBytesArray((SPIRead(REG_MODEM_CONFIG_2, 1).get(0) & 0x0F) | ((spreadingFactor << 4) & 0xF0));
```

Comment:
As per the python repo, this line & the line below it are outside of this else statement.